### ISSUE TITLE
Litematica not pasting schematics on a LAN world

### ISSUE THREAD

**Issue by Temastp** [maruohon/litematica#656]
Im currently playing with my brother on a LAN World (1.19.2)
When my brother left i closed the LAN World after that, i used litematica to paste some automated farms.
It didnt work it only says "Paste Schematic, Remaining Chunks (6)- i know the number can be different but every schematic doesnt work.
I created a new world and all schematics i had worked

I dont know what i should do now

**Comment by maruohon** [comment]
Assuming you are near those missing chunks so that they are loaded, then if the task just won't run, it's probably because your game is already running near or above 50 ms/tick. The Litematica tasks try to keep below 50 MSPT to not drop the game speed below the normal 20 TPS, so if the game is already lagging to begin with, then the tasks will never run.

Check your MSPT and try to get rid of some lag sources (from the chunks that are loaded while you try to paste stuff). Since it's basically single player, you can also try lowering the render distance (or I guess it's nowadays actually the separate simulation distance) while you paste things, if there is something laggy kinda nearby loaded.

**Comment by Temastp** [comment]
Where do i check the mspt?

And yeah my the world is pretty laggy, i see animals like moving a bit then pausing, is there a way to fix that?
Because all i built was an underground base with many lanterns to keep the mobs away, but there is not a lag machine or something like that

**Comment by maruohon** [comment]
Well the first step to getting rid of lag is to install the Lithium mod, if you don't have it already. And then also the Starlight mod helps a ton with light update related freezes and lag. And Sodium would help with client side rendering performance.

* https://modrinth.com/mod/lithium
* https://modrinth.com/mod/sodium
* https://www.curseforge.com/minecraft/mc-mods/starlight

To see the MSPT (ms per tick) value, in single player in the recent MC versions you can just see it on the F3 screen:
![mspt_f3](https://user-images.githubusercontent.com/2331050/213916844-72551c6b-a5ef-467c-a360-87130f2e5fc6.png)

Another alternative is using the MiniHUD mod, and enabling the `infoServerTPS` info line. This would also work with Carpet mod on the server side in multiplayer, or in upcoming versions with Servux on the server side to send the data.
![mspt_minihud](https://user-images.githubusercontent.com/2331050/213916880-81fcb5e4-66d4-4682-b31d-6658dfa33289.png)

Or the third option is using the `/log tps` command from the Carpet mod (either FabricCarpet or QuickCarpet) if the server has the mod installed. Then you see the value in the player list footer (MiniHUD can also read this value, if you are on a server that has Carpet mod):
![mspt_carpet](https://user-images.githubusercontent.com/2331050/213916970-139b9535-0c60-4fd0-83c1-6779765ccddb.png)

**Comment by Temastp** [comment]
Ok thanks, i fixed the lag by disabling some datapacks

**Comment by maruohon** [comment]
Yes they can. I have no idea what that datapack does and how, but some data packs that runs tons of complex commands all the time can certainly make a notable performance hit.

There is the built-in `/profiler` command that could give some clues, or you could try the Spark profiler: https://spark.lucko.me/

But probably the easiest and would be the `/tick health` command from [Carpet mod](https://www.curseforge.com/minecraft/mc-mods/carpet) - it would give you a simple overview of how much processing time blocks, entities etc. take in your world.
